packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1414,58 +1414,64 @@ class _TimePickerInputState extends State<_TimePickerInput> {
                 const SizedBox(width: 12.0),
               ],
-              Expanded(child: Column(
-                crossAxisAlignment: CrossAxisAlignment.start,
-                children: <Widget>[
-                  const SizedBox(height: 8.0),
-                  _HourMinuteTextField(
-                    selectedTime: _selectedTime,
-                    isHour: true,
-                    style: hourMinuteStyle,
-                    validator: _validateHour,
-                    onSavedSubmitted: _handleHourSavedSubmitted,
-                    onChanged: _handleHourChanged,
-                  ),
-                  const SizedBox(height: 8.0),
-                  if (!hourHasError && !minuteHasError)
-                    ExcludeSemantics(
-                      child: Text(
-                        MaterialLocalizations.of(context).timePickerHourLabel,
-                        style: theme.textTheme.caption,
-                        maxLines: 1,
-                        overflow: TextOverflow.ellipsis,
-                      ),
+              Expanded(
+                child: Row(
+                  // Hour/minutes should not change positions in RTL locales.
+                  textDirection: TextDirection.ltr,
+                  children: <Widget>[
+                    Expanded(child: Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: <Widget>[
+                        const SizedBox(height: 8.0),
+                        _HourTextField(
+                          selectedTime: _selectedTime,
+                          style: hourMinuteStyle,
+                          validator: _validateHour,
+                          onSavedSubmitted: _handleHourSavedSubmitted,
+                          onChanged: _handleHourChanged,
+                        ),
+                        const SizedBox(height: 8.0),
+                        if (!hourHasError && !minuteHasError)
+                          ExcludeSemantics(
+                            child: Text(
+                              MaterialLocalizations.of(context).timePickerHourLabel,
+                              style: theme.textTheme.caption,
+                              maxLines: 1,
+                              overflow: TextOverflow.ellipsis,
+                            ),
+                          ),
+                      ],
+                    )),
+                    Container(
+                      margin: const EdgeInsets.only(top: 8.0),
+                      height: _kTimePickerHeaderControlHeight,
+                      child: _StringFragment(timeOfDayFormat: timeOfDayFormat),
                     ),
-                ],
-              )),
-              Container(
-                margin: const EdgeInsets.only(top: 8.0),
-                height: _kTimePickerHeaderControlHeight,
-                child: _StringFragment(timeOfDayFormat: timeOfDayFormat),
+                    Expanded(child: Column(
```

Comment:
In the previous change, the Expand child was on the subsequent line. It's a little easier to read that way.

Reply:
Done